src/v2i-hub/TimPlugin/src/TimPlugin.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * TimPlugin.h
+ *
+ *  Created on: October 25, 2017
+ *      Author: zinkg
+ */
+
+#ifndef TIMPLUGIN_H_
+#define TIMPLUGIN_H_
+
+#include <iostream>
+#include <map>
+#include <mutex>
+#include <sstream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <thread>
+#include <time.h>
+#include <sys/time.h>
+#include <atomic>
+#include <tmx/tmx.h>
+#include <tmx/IvpPlugin.h>
+#include <tmx/messages/IvpJ2735.h>
+#include <GeoVector.h>
+#include <boost/date_time.hpp>
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <iomanip>
+#include <ctime>
+
+
+
+#include "PluginUtil.h"
+
+
+#include "PluginClient.h"
+
+#include <ApplicationMessage.h>
+#include <ApplicationDataMessage.h>
+
+#include <tmx/j2735_messages/BasicSafetyMessage.hpp>
+#include <tmx/messages/auto_message.hpp>
+
+using boost::property_tree::ptree;
+
+using namespace std;
+
+using namespace tmx;
+using namespace tmx::utils;
+using namespace tmx::messages;
+
+
+#define INPUTSREADY "Have Map/Spat/Veh"
+#define INGRESSREGION "Ingress Region"
+#define EGRESSREGION "Egress Region"
+
+
+
+struct ZoneInfo
+{
+	// ID of the zone the vehicle is in.  0 indicates no zone.
+	int ZoneId;
+
+	// The current speed of the vehicle in MPH.
+	float Speed_mph;
+
+	// The time the ZoneId was last updated.
+	// Note that if a 0 is received for ZoneId, this time will not be set unless
+	// it has been longer than a configurable duration.
+	uint64_t LastUpdateTime;
+};
+
+namespace TimPlugin {
+
+
+class TimPlugin: public PluginClient {
+public:
+	TimPlugin(std::string);
+	virtual ~TimPlugin();
+	int Main();
+protected:
+
+	void UpdateConfigSettings();
+
+	// Virtual method overrides.
+	void OnConfigChanged(const char *key, const char *value);
+	//void OnMessageReceived(IvpMessage *msg);
+	void OnStateChange(IvpPluginState state);
+
+	// Message handler functions
+	void HandleBasicSafetyMessage(BsmMessage &msg, routeable_message &routeableMsg);
```

Comment:
Remove methods that are not needed like BSM processing and vehicle in zone check.

Reply:
Addressed